### DISCUSSION TITLE
Best way to model many 2 many relations?

### DISCUSSION THREAD

**Discussion by mindreframer** [pocketbase/pocketbase#40] [Q&A] [answered]
Very interesting project, really appreciate the effort the open source it. 

One question, that might be tricky considering how young this code base: are many-2-many relations supported? Like tags <-> posts. 

Currently multiple values relations are stored as a single JSON array with IDs, this does not work for more complex data requirements. 

Is support for this on the roadmap?

**Comment by ganigeorgiev** [comment, marked as answer]
Currently the relation field supports one-to-one and one-to-many relations.

For many-2-many relation, I'd recommend to create a junction collection with 2 relation fields mapped to the tags and posts collections.

But you are correct that internally we are storing miltiple values as json array. I don't think this is a huge limitation, but I guess it depends on your use case. Could you elaborate more on your data requirements?

**Reply by ganigeorgiev** [reply]
Also keep in mind that PocketBase auto resolve nested search queries, so its OK to query for your tags and posts through the junction collection, eg. `tags.title = "test"`

**Reply by mindreframer** [reply]
Sorry for the late reply, somehow I missed the email notification. A manually maintained junction collection seems like an OK workaround, I would have to play with it. Also it could be useful for other users to have a central docs page with all possible modelling use cases, Entgo.io does it quite well - https://entgo.io/docs/schema-edges

Thanks for the quick response!
